src/main/java/hudson/plugins/warnings/parser/IarParser.java

```diff
@@ -14,27 +14,30 @@
  *
  * @author Claus Klein
  * @author Ulli Hafner
+ * @author Kay van der Zander
  */
 @Extension
 public class IarParser extends RegexpLineParser {
     private static final long serialVersionUID = 7695540852439013425L;
-
-    private static final String IAR_WARNING_PATTERN =
-    "^(?:\\[.*\\]\\s*)?\\\"?(.*?)\\\"?(?:,|\\()(\\d+)(?:\\s*|\\)\\s*:\\s*)(Error|Remark|Warning|Fatal error)\\[(\\w+)\\]: (.*)$";
-
+    private static int GROUP_NUMBER = 5;
+    
+    // search for: Fatal Error[Pe1696]: cannot open source file "c:\filename.c"
+    // search for: c:\filename.h(17) : Fatal Error[Pe1696]: cannot open source file "System/ProcDef_LPC17xx.h"
+    private static final String IAR_WARNING_PATTERN = 
+"((\\[exec\\] )?(.*)\\((\\d+)\\)?.*)?(Fatal [Ee]rror|Remark|Warning)\\[(\\w+)\\]: (.*(\\\".*(c|h)\\\")|.*)";
     /**
      * Creates a new instance of {@link IarParser}.
      */
     public IarParser() {
         super(Messages._Warnings_iar_ParserName(),
                 Messages._Warnings_iar_LinkName(),
                 Messages._Warnings_iar_TrendName(),
-                IAR_WARNING_PATTERN, true);
```

Comment:
If you remove the true, then the method isLineInteresting is **not** invoked anymore!